lib/backend/syncersv1/updateprocessors/globalnetworkpolicyprocessor.go

```diff
@@ -43,16 +45,24 @@ func convertGlobalNetworkPolicyV2ToV1Value(val interface{}) (interface{}, error)
 	if !ok {
 		return nil, errors.New("Value is not a valid GlobalNetworkPolicy resource value")
 	}
-	return convertGlobalPolicyV2ToV1Spec(v3res.Spec)
-}
 
-func convertGlobalPolicyV2ToV1Spec(spec apiv3.GlobalNetworkPolicySpec) (*model.Policy, error) {
+	spec := v3res.Spec
+	selector := spec.Selector
+
+	nsSelector := spec.NamespaceSelector
+	if nsSelector != "" {
+		nsSelector = strings.Replace(nsSelector, "all()", "has(projectcalico.org/namespace)", -1)
```

Comment:
I think these two lines should be switched, right?

otherwise we'll end up with `has(pcns.projectcalico.org/namespace)` instead of `has(projectcalico.org/namespace)`?

Reply:
Fixed.. tho none of the existing tests caught this :-/